fred2/voiceactingmanager.cpp

```diff
@@ -943,3 +951,176 @@ void VoiceActingManager::OnBnClickedIncludeSender()
 
 	UpdateData(FALSE);
 }
+
+void VoiceActingManager::OnCopyPersonas(bool messages_to_ships)
+{
+	char sender_buf[NAME_LENGTH];
+	int sender_shipnum;
+
+	SCP_unordered_set<int> already_assigned;
+	SCP_string inconsistent_copy_msg;
+
+	// go through all messages in the mission
+	for (int i = 0; i < Num_messages - Num_builtin_messages; i++)
+	{
+		auto message = &Messages[i + Num_builtin_messages];
+
+		// find whoever sent this message
+		get_valid_sender(sender_buf, NAME_LENGTH, message, &sender_shipnum);
+
+		// if it's a ship, copy the persona
+		if (sender_shipnum >= 0)
+		{
+			auto sender_shipp = &Ships[sender_shipnum];
+			auto persona_to_copy = (messages_to_ships) ? message->persona_index : sender_shipp->persona_index;
+
+			// don't copy None, and only copy wingman personas
+			if (persona_to_copy >= 0 && (Personas[persona_to_copy].flags & PERSONA_FLAG_WINGMAN))
```

Comment:
I take it the reason only wingmen personas are copied here (and deleted for later on) is b/c there are many other checks throughout message code that limit things to just wingmen personas?

Reply:
Well, that may be related. My reason is that there is only one Terran Command persona, so mission designers could conceivably use it for any Terran head animation, or all of them, in one mission.  The persona isn't standardized in the way that the wingmen are.

Reply:
Ah, gotcha. There's also the possibility of mods that have more than one command personas for a given species, but I take it you have thought of that already.

Reply:
Yep.  If mods created multiple Command personas and used them for different missions, and if mods were consistent with the use of head animations, then it would be appropriate to include them.  But occasionally mods will use a Command persona with a different head animation, or assign the Command persona to a ship.  That would result in undesirable side effects if the voice manager dialog processed those personas.

Reply:
Makes sense, thanks for the extra explanation.